pkg/configuration/configuration.go

```diff
@@ -484,9 +577,99 @@ func (ev *extendedViper) GetAllKeysThatContainValues(key string) []string {
 }
 
 func (ev *extendedViper) GetKeyType(key string) KeyType {
-	if strings.HasPrefix(strings.ToLower(key), "snyk_") {
-		return EnvVarKeyType
+	// check for supported env vars
+	for _, envVar := range ev.supportedEnvVars {
+		if strings.EqualFold(key, envVar) {
+			return EnvVarKeyType
+		}
+	}
+
+	// check for supported prefixes
+	for _, prefix := range ev.supportedEnvVarPrefixes {
+		if strings.HasPrefix(strings.ToLower(key), strings.ToLower(prefix)) {
+			return EnvVarKeyType
+		}
 	}
 
 	return UnspecifiedKeyType
 }
+
+// AutomaticEnv wraps Viper's AutomaticEnv and allows us to check if this has been set or not.
+// If AutomaticEnv is enabled, SetSupportedEnvVars and SetSupportedEnvVarPrefixes is disabled.
+func (ev *extendedViper) AutomaticEnv() {
+	ev.mutex.Lock()
+	defer ev.mutex.Unlock()
+
+	ev.viper.AutomaticEnv()
+	ev.automaticEnvEnabled = true
+}
+
+func (ev *extendedViper) GetAutomaticEnv() bool {
+	if ev.mutex.TryLock() {
```

Comment:
TryLock is making the situation worse, as a will also fail if someone else locked the mutex, which is the opposite of what we want. Let's sync on a solution.